sdk/src/main/java/com/mapbox/maps/ResourceOptionsManager.kt

```diff
@@ -24,20 +24,56 @@ class ResourceOptionsManager(
    * Static variables and methods.
    */
   companion object {
-    private lateinit var default: ResourceOptionsManager
+    private var default: ResourceOptionsManager? = null
+    internal fun getTokenResId(context: Context): Int = context.resources.getIdentifier(
+      MAPBOX_ACCESS_TOKEN_RESOURCE_NAME,
+      "string",
+      context.packageName
+    )
+
+    /**
+     * Convenience function to remove the default instance. Calling `getDefault` again will re-create the default instance.
+     */
+    fun destroyDefault() {
+      default = null
+    }
 
     /**
      * The default shared instance with default resource options.
+     *
+     * @param context the application context
+     * @defaultToken the default token will be set to the default [ResourceOptionsManager] instance.
+     * If set to null, will reuse the previously set one or search for an access token in the application resources for the first time of init.
+     * It will throw [MapboxConfigurationException] if there's no token found.
      */
     @Synchronized
-    fun getDefault(context: Context): ResourceOptionsManager {
-      if (!this::default.isInitialized) {
-        default = ResourceOptionsManager(
-          ResourceOptions.Builder().applyDefaultParams(context)
-            .build()
-        )
+    fun getDefault(context: Context, defaultToken: String? = null): ResourceOptionsManager {
+      // Apply the user setting token as the default token.
+      defaultToken?.let { token ->
+        default?.let {
+          // Reuse the config from previous default and only update the token
+          val builder = it.resourceOptions.toBuilder().accessToken(token)
+          return ResourceOptionsManager(builder.build()).also { default = it }
```

Comment:
Shall we make `update` return the updated ResourceOptionsManager instance and do the following?
```suggestion
          return it.update { accessToken(token) }
```

Reply:
The return type of `update` method is Unit, we can't return it directly.

Reply:
we can make update return the updated `ResourceOptionsManager` instance, if it makes sense.